native/src/main/java/io/ballerina/stdlib/kafka/nativeimpl/consumer/Poll.java

```diff
@@ -122,10 +128,11 @@ private static RecordType getRecordType(BTypedesc bTypedesc) {
         RecordType recordType;
         if (bTypedesc.getDescribingType().isReadOnly()) {
             recordType = (RecordType) getReferredType(((IntersectionType) getReferredType(((ArrayType) getReferredType(
-                    bTypedesc.getDescribingType())).getElementType())).getConstituentTypes().get(0));
+                    TypeUtils.getImpliedType(bTypedesc.getDescribingType()))).getElementType()))
```

Comment:
Since `getImpliedType` handles referred type cases as well you may be able to remove some of `getReferredType` cases as well please check